guevaraTorres/assignments/slidingwindow/lc340/lc340.js

```diff
@@ -4,28 +4,106 @@
 
 // You can assume that K is less than or equal to the length of the given string.
 
+/**
+ * 
+ *The window will need to extend while checking to see if there are more than k distinct characters.
+ * When that occurs the window will move up once the earliest character type is removed.
+ * Determining the longest substring(maxLength) while the window slides forward.
+ * 
+ * variables: windowStart, maxLength, characterFrequency, windowEnd
+ * 
+ * edge cases: null, k=0 or ""
+ */
```

Comment:
Nice high-level approach and edge cases. Don't forget to write what should be returned for each edge case.